Remove NormalizationRunnerFactory

andriikorotkov opened the pull request:
## What
Removed **NormalizationRunnerFactory.class** and updated all places where it using. Updated **DefaultNormalizationRunner.class** and changed `NormalizationRunnerFactory#create` to `new DefaultNormalizationRunner`. Updated **DestinationAcceptanceTest** and added methods for read inaormation about normalization and DBT from definition file. Updated **check_images_exist.sh** to get information from `destination_definition.yaml` file instead of **NormalizationRunnerFactory** class

## How
*Describe the solution*

## Recommended reading order
1. `DefaultNormalizationRunner.java`
2. `DestinationAcceptanceTest.java`
3. `check_images_exist.sh`

## 🚨 User Impact 🚨
Removed **NormalizationRunnerFactory.class**. Thereby we separate the normalization from the platform.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

andriikorotkov committed:
updated IntegrationLauncherConfig.yaml and added to this class suportDBT and normalizationImage fields. Added to the GenerateInputActivityImpl and TemporalClient classes code parts for read destination_definition.yaml and get suportDBT and normalizationImage fields. Added logging and comparing normalization images from NormalizationRunnerFactory and destination_definition.yaml

andriikorotkov committed:
updated minor remarks

andriikorotkov committed:
updated minor remarks

andriikorotkov committed:
fixed minor remarks

andriikorotkov committed:
added normalization data to the tests

andriikorotkov committed:
fixed minor remarks

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/18232

andriikorotkov committed:
removed NormalizationRunnerFactory

andriikorotkov committed:
fixed remarks

andriikorotkov committed:
Merge branch 'akorotkov/18232' of github.com:airbytehq/airbyte into akorotkov/remove_factory

 Conflicts:
	airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/18232

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/18232

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/18232

andriikorotkov committed:
Merge branch 'akorotkov/18232' of github.com:airbytehq/airbyte into akorotkov/remove_factory

andriikorotkov committed:
fixed remarks

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/remove_factory

 Conflicts:
	airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java
	airbyte-commons-worker/src/test/java/io/airbyte/workers/normalization/NormalizationRunnerFactoryTest.java
	airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/DbtJobOrchestrator.java
	airbyte-container-orchestrator/src/main/java/io/airbyte/container_orchestrator/NormalizationJobOrchestrator.java
	airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/DestinationAcceptanceTest.java
	airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/DbtTransformationActivityImpl.java
	airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/NormalizationActivityImpl.java

andriikorotkov committed:
fixed remarks

andriikorotkov committed:
updated acceptance tests

andriikorotkov committed:
updated acceptance tests

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/remove_factory

andriikorotkov committed:
updated check_images_exist.sh script

andriikorotkov committed:
updated method for get normalization image name for destination acceptance test

andriikorotkov committed:
fixed code style

andriikorotkov committed:
Merge branch 'master' of github.com:airbytehq/airbyte into akorotkov/remove_factory

andriikorotkov committed:
fixed code style and removed tests data

Commit:
🪟 🎉 Add options to tooltip of oneOf properties (#20015)

* add options to tooltip
* update story
* review comments

xpuska513 committed:
fix: comment downtime sync test and restore helm ac tests (#19944)

* fix: comment downtime sync test and restore helm ac tests

* Update as per review

* fix: fix class reference in test

* Comment out the test

* Import junit.Disabled

* Use default image

* Display logs of k3s service

* Use systemd instead of cgroup for kubelet

* Update airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/ContainerOrchestratorAcceptanceTests.java

Co-authored-by: Topher Lubaway <asimplechris@gmail.com>

* Update ContainerOrchestratorAcceptanceTests.java

* Update gradle.yml

Co-authored-by: Topher Lubaway <asimplechris@gmail.com>

andriikorotkov committed:
updated JobErrorReporterTest.java

jeromedoucet committed:
:bug: Source ElasticSearch: avoid too_long_frame_exception (#18134)

* Fix: (elasticsearch source) avoid too_long_frame_exception

batch the queries on mapping with a arbitrary (but reasonable) chunk size
to avoid reaching the 4096 bytes limits url size.

* bump connector

* auto-bump connector version

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

andriikorotkov committed:
updated JobErrorReporterTest.java

MaxiChef committed:
Fix formatting typo octavia list` command" (#19213)

pedroslopez commented on airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:
```diff
@@ -55,31 +54,17 @@ public class DefaultNormalizationRunner implements NormalizationRunner {
+
+  private String getDestinationTypeFromNormalizationImage(final String normalizationImage) {
+    String imageNameWithoutTag = normalizationImage.contains(":") ? normalizationImage.split(":")[0] : normalizationImage;
+    return "airbyte/normalization".equalsIgnoreCase(imageNameWithoutTag) ? "bigquery"
```
Looking at the previous logic, it seems like this doesn't always mean "bigquery". Postgres seems to use the same base image:

https://github.com/airbytehq/airbyte/blob/b452d53e9730aa8bbcc1b1790180c74a38ff75ed/airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java#L35-L36

andriikorotkov replied:
@pedroslopez Then I see two ways to solve this issue:
1) I can parse the name of the docker image (not normalization) and take the definition type.
2) I can add an additional field to the `destination_definition` file (and also to the `actor_definition` table) and read the type from there

The second way seems more logical to me. What do you say to that? Maybe you see some other option how best to get this type?

pedroslopez reviewed (commented):
Thanks andrii! Left a coupe questions - my main concern at the moment is how we're determining the destination type but I'm also not really sure what that's used for. Since I'm not as familiar with normalization or the standard destination acceptance tests I'd love to also get some input from someone on the dbs team, perhaps @edgao.

pedroslopez commented on tools/bin/check_images_exist.sh:
```diff
@@ -82,16 +82,18 @@ checkPlatformImages() {
-  image_version=$(cat $factory_path | grep 'NORMALIZATION_VERSION =' | cut -d"=" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
-  echo -e "$blue_text""Checking normalization images with version $image_version exist...""$default_text"
-  VERSION=$image_version 
+  normalization_image_versions=$(cat $definition_file_path | grep 'normalizationTag:' | cut -d":" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
+  IFS=' ' read -r -a array <<< "$normalization_image_versions"
```
Since we have normalization image and tag as part of the definition file, I feel like this check can be further simplified to be more like the `checkConnectorImages` function, looking up the normalization image and normalization tag pair to see if it exists on docker hub. The current aproach requires a docker-compose file only to have the image names, which doesn't seem necessary anymore. Feel free to make a separate issue / different PR for updating this script though.

andriikorotkov replied:
This is a good point. I agree with you, but I suggest leaving these changes, and in the next pull request update them as you described. Because the script will not find the required class and part of the checks will not pass.

mdibaiee committed:
source-mailchimp: set primary_key for EmailActivity (#19023)

* source-mailchimp: set primary_key for EmailActivity

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

sarafonseca-123 committed:
🐛 Source RD Station Marketing: Fix stream analytics conversions (#18826)

* Fix stream analytics conversions on RD Station Marketing

* update connector's version

* Fix PR number on changelog

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
Enable PMD rule to prevent == string comparisons (#20080)

Zawar92 committed:
Source RKI-Covid - add history data, incidence rate, cases, death, hospitalization streams (#17607)

* added streams states and states/history/cases

* fix: remove changes to formatting for other connectors (again)

* auto-bump connector version

Co-authored-by: sajarin <sajarindider@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
remove un-used update/delete custom connector api (#20034)

* remove un-used update/delete custom connector api

* also remove create path

* more tests to fix

marcelopio committed:
octavia-cli: Added flag to control whether to reset the streams or not (#18337)

* Added flag on octavia to control wheter to reset the streams or not

* Restore test_api_http_headers

* Going back to reseting by default

akashkulk committed:
JDBC Sources Integration Test migration follow-up: Switch additional tests to use system stubs (#20040)

* Migrating remaining tests

* More tests

* set env vars before querying them; use stream state everywhere?

* fix strirct encrypt expected spec

* runRead after workerConfigs exist

* format

Co-authored-by: Edward Gao <edward.gao@airbyte.io>

terencecho committed:
Revert "octavia-cli: Added flag to control whether to reset the streams or not (#18337)"

This reverts commit 4f9b66071e0325a646f3bf6ec3608273ab963a93.

Commit:
Add missing "unknown" reason (#20096)

malikdiarra committed:
Add test on `failureOrigin` enum conversion (#20101)

Commit:
Add testid to error view (#20095)

teallarson committed:
🪟 🎨 Row highlighting for new streams table (#19449)

* initial pass and cleanup

* merge cleanup

* get dropdowns colored correctly also

* colors for modificationIcon

* icon alignment

* testing skeleton

* move calculation to a hook

* update hover styles on rows with changes

* add testing

Co-authored-by: Krishna (kc) Glick <krishnaglick@users.noreply.github.com>

* clarify naming

* add mixin, fix selected added/removed rows

* WIP test icons using snapshots

* fix snapshots, tests for catalog tree table row hook

* fix order of logic for disabled rows (disabled trumps everything)

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.test.tsx

Co-authored-by: Edmundo Ruiz Ghanem <168664+edmundito@users.noreply.github.com>

* move icon to its own component, clean up tests

* tidy up test linting disables

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.tsx

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.tsx

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.tsx

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.test.tsx

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

* Update airbyte-webapp/src/components/connection/CatalogTree/next/useCatalogTreeTableRowProps.test.tsx

Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

* clean up tests

* make  path-select dropdown full column width

* cleanup from rebase

Co-authored-by: Krishna (kc) Glick <krishnaglick@users.noreply.github.com>
Co-authored-by: Edmundo Ruiz Ghanem <168664+edmundito@users.noreply.github.com>
Co-authored-by: Krishna (kc) Glick <krishna@airbyte.io>

erohmensing committed:
Add README info about using your local CDK locally and in docker (#20082)

* Add README info about installing the local CDK locally and in Docker

* Add '|| true' to validate_dockerignore()

* Update airbyte-cdk/python/README.md

Co-authored-by: Sherif A. Nada <snadalive@gmail.com>

Co-authored-by: Sherif A. Nada <snadalive@gmail.com>

edgao commented on airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:
```diff
@@ -55,31 +54,17 @@ public class DefaultNormalizationRunner implements NormalizationRunner {
+    this.destinationType = getDestinationTypeFromNormalizationImage(normalizationImage);
+  }
+
+  private String getDestinationTypeFromNormalizationImage(final String normalizationImage) {
+    String imageNameWithoutTag = normalizationImage.contains(":") ? normalizationImage.split(":")[0] : normalizationImage;
+    return "airbyte/normalization".equalsIgnoreCase(imageNameWithoutTag) ? "bigquery"
+        : imageNameWithoutTag.replace("airbyte/normalization-", "");
```
it's an [airbyte thing](https://github.com/airbytehq/airbyte/blob/4bef54d1b9873f2609a4ca8064d3607de9f5a6b1/airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py#L40), but we do need it explicitly for normalization. It controls how we generate various SQL statements (e.g. maximum table name length, model structure, etc).

Once we remove the strict-encrypt variants, this should be 1:1 with the destination image names (e.g. image name `destination-bigquery` -> destination type `bigquery`). Not sure if that's actually better though.

(destinaiton-bigquery-denormalized doesn't actually support normalization, so its inclusion in this map is just weird historical wrongness)

edgao commented on airbyte-commons-worker/src/test/java/io/airbyte/workers/normalization/DefaultNormalizationRunnerTest.java:
```diff
@@ -82,7 +85,7 @@ void setup() throws IOException, WorkerException {
 
     when(processFactory.create(AirbyteIntegrationLauncher.NORMALIZE_STEP, JOB_ID, JOB_ATTEMPT, jobRoot,
-        NormalizationRunnerFactory.BASE_NORMALIZATION_IMAGE_NAME, false, files, null,
+        "airbyte/normalization", false, files, null,
```
nitpick: does this intentionally not use the new `NORMALIZATION_IMAGE` constant?

andriikorotkov replied:
Oh! Thanks for noticing. I updated the test first, then added the constant, and didn't add the change to the test. =)

terencecho committed:
octavia-cli: undue skip_reset flag revert and update integration tests  (#20093)

* Revert "Revert "octavia-cli: Added flag to control whether to reset the streams or not (#18337)""

This reverts commit da0d9a7984807c619a8242b304e6493bd71168ea.

* Update octavia cli integration tests

edgao commented on airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/DestinationAcceptanceTest.java:
```diff
@@ -121,6 +123,28 @@ public abstract class DestinationAcceptanceTest {
+
+  private Optional<StandardDestinationDefinition> getOptionalDestinationDefinitionFromProvider(final String imageNameWithoutTag) {
+    try {
+      LocalDefinitionsProvider provider = new LocalDefinitionsProvider(LocalDefinitionsProvider.DEFAULT_SEED_DEFINITION_RESOURCE_CLASS);
```
this means that the destination acceptance tests now depend on the destination_definitions.yaml file - is that ok? (I don't know if this explicitly breaks anything, just want to doublecheck that we're not introducing any exciting circular dependencies. E.g. if I have a PR that modifies some behavior in `destination-foo` _and_ switches it from `normalization` to `normalization-foo`, is that still easy to do?)

andriikorotkov replied:
Yes, you are right that for destination acceptance tests we end up depending on `destination_definitions.yaml`. As for your example, switches `normalization` to `normalization-foo` for `destination-foo` is very easy. Only one field for destination-foo in the `destination_definitions.yaml` file will need to be updated. This is the easiest way to integrate new logic into destination acceptance tests.

edgao commented on airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/DestinationAcceptanceTest.java:
```diff
@@ -96,7 +99,6 @@
 public abstract class DestinationAcceptanceTest {
 
   private static final Random RANDOM = new Random();
-  private static final String NORMALIZATION_VERSION = "dev";
```
are we no longer overriding the version with `dev`? (if yes: that's a problem, because it will be _really_ difficult to test a PR that modifies a destination connector _and_ normalization)

andriikorotkov replied:
Thank you for your comments. This is my shortcoming. Instead of the dev version, I started taking the latest version from the definition file. Changed it and double checked that everything works fine

edgao commented on tools/bin/check_images_exist.sh:
```diff
@@ -82,16 +82,18 @@ checkPlatformImages() {
-  image_version=$(cat $factory_path | grep 'NORMALIZATION_VERSION =' | cut -d"=" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
-  echo -e "$blue_text""Checking normalization images with version $image_version exist...""$default_text"
-  VERSION=$image_version 
+  normalization_image_versions=$(cat $definition_file_path | grep 'normalizationTag:' | cut -d":" -f2 | sed 's:;::' | sed -e 's:"::g' | sed -e 's:[[:space:]]::g')
```
how hard would it be to install `yq` instead of doing this? (nonblocking)

andriikorotkov replied:
I don't think it will take long. But I suggest leaving it as it is. I plan to return to this file anyway, at the end of the platform separation and normalization work.

edgao reviewed (commented):
:hype: this is exciting!

lmossman committed:
:window: :wrench: Generify GroupControls to make it more reusable (#20004)

* generify GroupControls more

* undo radius change

* use spacing variable

edgao committed:
Source Postgres: Fix not being able to configure wal2json plugin (unpublished) (#19985)

* switch const to default

* update expected specs

* try updating system-stubs?

* Revert "try updating system-stubs?"

This reverts commit 0311a5a1d15eedb83e4c9b82ecf33f4c264ae60d.

* switch to system stubs

* changelog

* fix more tests

terencecho committed:
Validate connection id before sync (#20107)

* Validate connection id before sync

* Propogate changes

akashkulk committed:
Move handling TIMESTAMP_WITH_TIMEZONE to common JdbcSourceOperations (#19860)

* Refactor

* Fix build failures

alovew committed:
refresh before syncs when feature flag is on (#19888)

* refresh before syncs when feature flag is on

andriikorotkov commented on airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:
```diff
@@ -99,12 +84,12 @@ public boolean configureDbt(final String jobId,
     if (Strings.isNullOrEmpty(gitRepoBranch)) {
       return runProcess(jobId, attempt, jobRoot, files, resourceRequirements, "configure-dbt",
-          "--integration-type", destinationType.toString().toLowerCase(),
+          "--integration-type", destinationType.toLowerCase(),
```
As far as I can see, this type is used for which dialect to use when normalizing. I think Ed's comment explains it better.
![2022-12-06_11-57](https://user-images.githubusercontent.com/88329385/205880249-e161f2a1-ef7f-4ebc-9a53-4be8e86317e3.png)

Commit:
fix type number in schema (#20068)

Commit:
warn on navigating away from unsaved connector form with error (#20072)

andriikorotkov committed:
fixed remarks

Commit:
🪟 🔧 Add comment about feature/experiment usage (#20123)

maxi297 committed:
[ISSUE #20086] fix precommit pyproject-flake8 dependency issue (#20087)

Commit:
🪟 🐛 Fix tag input styles (#20020)

* fix input styles
* pointer cursor for the remove buttons

alafanechere committed:
SAT: Capture configuration updates from connectors' control messages (#19979)

erohmensing committed:
Add gradle --scan and :tasks to gradle documentation (#20102)

roman-yermilov-gl committed:
airbyte-cdk offset pagination strategy: page_size to be interpolated … (#19646)

* airbyte-cdk offset pagination strategy: page_size to be interpolated string
* airbyte-cdk offset pagination strategy: bump version

DoNotPanicUA committed:
Extend template for the Java JDBC template by SAT SPEC (#19660)

* extend template for the Java JDBC template by SAT SPEC

* extend example source scaffold-java-jdbc

* review upd

* add other SAT tests to the template

* review upd

* comment non-spec tests

* Revert "comment non-spec tests"

This reverts commit 8c78049fd9a5fee54c7715289cd1ded0fdf05e0f.

* Revert "add other SAT tests to the template"

This reverts commit c15b4dcd87863bbf235045538e93de355fec172c.

* review

* review

* typo

* typo

* new line

edmundito committed:
🪟 🎨 Auto-detect schema changes UI fixes (#20070)

* Update Review changes button to dark variant
Fix wording on schema change strings

* Disable pointer from label in EnabledControl when switch is disabled

* Fix hover state for connector links when in breaking or non-breaking change

edgao reviewed (approved):
lgtm from destinations POV; I'll defer to pedro about best approach for getting the `integration_type` param.

akashkulk committed:
Fix additional MySQL integration tests (#20119)

* Fix test : attempt 1

* Remove setting cloud

erica-airbyte committed:
Update mssql.md (#13663)

Co-authored-by: Amruta Ranade <11484018+Amruta-Ranade@users.noreply.github.com>

tybernstein committed:
Update Stripe Customer Stream with additional info (#19975)

Clarifies that the customer stream does not include deleted customers

sophia-wiley committed:
edits to managing airbyte cloud doc (#20108)

artem1205 committed:
🐛 Source Facebook Marketing: update next cursor value at read start (#20131)

* Source Facebook Marketing: update next cursor value at read start

* Source Facebook Marketing: update docs

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

roman-yermilov-gl committed:
airbyte-cdk: bump version (#20143)

oleksii-nikitchuk committed:
:bug: Source Kafka: fix missing data (#19587)

* Kafka Source - fix missing data

* auto-bump connector version

* update doc

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

alafanechere committed:
source-java-jdbc: use latest acceptance-test-config.yml format (#20144)

ambirdsall committed:
Add dbt cloud integration to default cloud features (#20105)

genzgd committed:
🐛 Destination ClickHouse: Update Normalization Docker File (#19573)

* Update ClickHouse normalization docker file

* bump destination and norm version

* auto-bump connector version

* update doc

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

gosusnp committed:
Revert "RecordSchemaValidator can resolve $ref schemas (#19625)" (#20113)

This reverts commit 56bfdaab6a1edddde15dc2f46e2e12d92ff21210.

edmundito committed:
🪟 🔧 Add auto-detect schema changes feature flag (#20035)

* Add auto-detect schema changes feature flag

* Update schema changes enabled in StatusCell

* Mock useIsAutoDetectSchemaChangesEnabled hook in tests

akashkulk committed:
MySQL Source : Check for `REQUIRE CLIENT` privilege in CDC (#20000)

* Test

* Check for REPLICATION CLIENT PRIVILEGE

* format

* undo format

* Fix test + comments

* Simplify checked function

* Bump up version + log

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

edmundito committed:
Move CatalogDiff modal from views to components (#20024)

evantahler committed:
Destination doc and warning updates (#20110)

* Doc updates

* Bigquery Denormalized

* bump faker for change

* ignore missing strict-encrypt connectors from ci check

* Apply suggestions from code review

Co-authored-by: Augustin <augustin@airbyte.io>

* Fix MD titles

Co-authored-by: Augustin <augustin@airbyte.io>

clnoll committed:
Show http method in the connector builder test panel request tab (#20109)

Show http method in the connector builder test panel request tab

sashaNeshcheret committed:
Postgres source: handle terminate connection exception with test. (#19887)

* Source postgres: catch termination connection PSQLException

* Source postgres: move common code to util method

* Source postgres: clean code

* Source postgres: review fixes and added unit tests

* Source postgres: clean code

* Source postgres: bump version

* Source postgres: format

* Source postgres: format

* Source postgres: bump version

* Source postgres: fix failing test and bump version

* Source mysql: make message visible for tests

* Source postgres: reformat import

Co-authored-by: VitaliiMaltsev <39538064+VitaliiMaltsev@users.noreply.github.com>
Co-authored-by: Akash Kulkarni <113392464+akashkulk@users.noreply.github.com>
Co-authored-by: Rodi Reich Zilberman <867491+rodireich@users.noreply.github.com>

rodireich committed:
Stop sync on a null value in a cursor column (#19889)

* Stop sync on a null value in a cursor column

* Fix quoting in query to solve a failing test

* Fix another escaping issue in query

* Fix failing test

* Fix failing tests

* Test view with null value cursor

* Improve error message

* bump dockerfile version and update note

* bump version to 1.0.31

* Fix failing test

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

evantahler committed:
`socat` gets more resources (#19953)

* `socat` gets more resources

* Move pod limits to ENV

* lint

maxi297 committed:
Issue 19733 cdk clarify config error message for config files (#20019)

* [ISSUE-19733] clarify error message when reading config files

* [ISSUE #19733] code review and adding validation for spec file as well

* [ISSUE #19733] updating typing of read_json_file

* [ISSUE #19733] fix flake8 error

* [ISSUE #19733] fix linting error

* [ISSUE #19733] remove breaking change

* [ISSUE #19733] bump airbyte cdk version

* [ISSUE #19733] add test for invalid json file on read_state

* [ISSUE #19733] bump version

edgao committed:
Destination Clickhouse: Publish strict encrypt variant (#20171)

* derp change

* bump version

ptiurin committed:
Source Firebolt: allow reading from views (#16583)

* Source Firebolt: allow reading from views

* setting cdk version

* Decimal is now enabled

* Bump sdk version to 0.12

* fix integration tests

* Disabling backwards compatibility test

* address comments:
- revert .pre-commit-config.yaml
- update versions manually under the seed file
- update the doc to reflect the PR link

* fix docs and bump source_specs.yaml

Co-authored-by: Yiyang Li <yiyangli2010@gmail.com>

Commit:
Use a separate node pool to run custom connector jobs (#19770)

* kubeprocessfactory flag

* plumbing through custom connector bit

* config, style, pmd

* fix more missing configs pass along

* fix sync config

* test fix

* add a flag to specify if we want to use a separate pool or not

* add missing micronaut configs

* make orchestrator job to run in custom pool too

* micronaut fix

* pass config to orchestrator

* fix test

* destination test fix

* PR comments fix

* style fix

* comment fix

* no checks on kubeprocess

* rename

evantahler committed:
lint CDK files (#20163)

* lint CDK files

* bump to rebuild file

* format again?

* remove double line

* fix spacing in template

* remove trailing space

lmossman committed:
[Connector Builder] upgrade orval and use connector manifest schema directly (#20166)

* upgrade orval and use connector manifest schema directly

* remove airbyte protocol from connector builder api spec

* generate python models from openapi change

girarda committed:
Alex/lowcode custom components docs (#20176)

* Update label

* reset

* update

* update

* update

* Update

* update

bgroff committed:
Remove Change Management section as it was outdated. Added Authentication section to clarify how to use the API in OSS. (#20167)

* Remove the Change Management section as it was outdated. Added Authentication section to clarify how to use the API in OSS.

Co-authored-by: swyx <shawnthe1@gmail.com>

alafanechere committed:
ci-credentials: update GSM secrets with updated configuration values (#20076)

Commit:
🪟 🎉 Add datepicker for date/date-time fields in connector form (#19678)

Co-authored-by: Tim Roes <tim@airbyte.io>
Co-authored-by: Lake Mossman <lake@airbyte.io>

artem1205 committed:
🐛 Source Facebook Marketing: fix fields permission error (#20165)

* Source Facebook Marketing: fix fields permission error

* Source Facebook Marketing: update docs

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
🪟🧹 Improve connector form types (#20057)

* improve some types

Commit:
fix types (#20187)

krishnaglick committed:
Added stylelintignore for generated coverage files. (#20097)

* Added stylelintignore for generated coverage files.

* Alex CR

alafanechere committed:
test/publish commands: always run update secrets (#20183)

Commit:
🪟 🎉 new streams table edit modal namespace and stream (#19713)

* Update new streams table to edit namespace and stream name options with modals

- Added modal components `Namespace` and `Stream name`
- Fixed bug with Tooltip (block-level element inside inline element)

LuliRib committed:
Source Bing Ads: add fields to schema (#20005)

* Update keyword_performance_report.json

* Update keyword_report.json

* Update keyword_report.json

deleted "metrics.historical_quality_score" which is not compatible with segments.device 
This was verified by the google ads team

* rollback google ads change

* bump connector version

* Update keyword_performance_report.json

added property additionalProperties: true to keyword field

* Update keyword_performance_report.json

* Update keyword_performance_report.json

I think now it´s what you asked.
Thank you

* Update keyword_performance_report.json

added "additionalProperties": true to the root of schema

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

juweins committed:
Docs: update list of supported streams in Freshdesk (#20125)

* Update list of supported streams #19890

* Update freshdesk.md

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

roman-yermilov-gl committed:
Source Close.com: migration to low-code (#18968)

* Source Close.com: migration to low-code

* Source Close.com: bump version; update changelog

* Source Close.com: documentation structure standartization

* Source Close.com: making connector to be prepared to beta

* Source Close.com: migrate SAT  config to new version

* Source Close.com: unit tests for custom components

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
regenerated lockfile with resolved & integrity (#20191)

evantahler committed:
Display Throughput and Progress Bar in Sync UI (#19193)

* Display Throughput and Progress Bar in Sync UI

* test for numberHelper

* use clear button

* noEstimate

* remove ?

* cleanup ternary

* clear up styles

* fixup color ref

* cleanup and test math

* pluralize

* Custom progress bar line

* remove padding on clear buttons

* Tim's suggestions

* size=xs

* Some cleanups

* Continue work (WIP)

* WIP

* WIP

* WIP

* Cleanup utils file

* Cleanup i18n

* Cleanup unused classes

* Cleanup more leftovers

* Remove debug data

Co-authored-by: Tim Roes <tim@airbyte.io>

teallarson committed:
🪟 🎉 Add auto-detect schema changes option to connection UIs (#19734)

* WIP option field NOTE: does not save update properly!

* cleanup, hide behind flag

* cleanup from rebase

* update snapshots

* update field description

* Addressing PR review comments

* fix bad variables

* useField after all

* pull touched and errors from form

* update snapshots (this field is hidden now in main ui)

* use hook instead of env var to check for feature

* use hook for validation schema creation too

Commit:
🪟 🐛 Fix test assertion in datepicker tests (#20188)

* fix test assertion and mock TZ

* fix typo

Commit:
🪟 🎉 Change default sync mode (#20126)

* Change default sync mode logic

* FIx jest tests

edmundito committed:
🪟 🧹 Improve and fix linting in webapp-e2e-tests (#20134)

* Apply similar eslint rules to the webapp project and fix outstanding issues

* add paths to e2e tsconfig (for webstorm)

sophia-wiley committed:
November release notes docs (#20199)

* added nov release notes

* edits

Commit:
Add custom connector flag for reset job (#20205)

* kubeprocessfactory flag

* plumbing through custom connector bit

* config, style, pmd

* fix more missing configs pass along

* fix sync config

* test fix

* add a flag to specify if we want to use a separate pool or not

* add missing micronaut configs

* make orchestrator job to run in custom pool too

* micronaut fix

* pass config to orchestrator

* fix test

* destination test fix

* PR comments fix

* style fix

* comment fix

* no checks on kubeprocess

* rename

* add flags for reset work

* test fix

grubberr committed:
CDK: offset_increment.page_size make as [int, InterpolatedString] (#20190)

Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

mfsiega-airbyte committed:
database migration to add column for field selection info (#20202)

octavia-squidington-iii committed:
Bump Airbyte version from 0.40.23 to 0.40.24 (#20206)

Co-authored-by: girarda <girarda@users.noreply.github.com>

DoNotPanicUA committed:
Enable full SAT for the Snowflake source (#20138)

* Enable full SAT for the snowflake

* readme + format

* Update airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml

Co-authored-by: Augustin <augustin@airbyte.io>

* Revert "Update airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml"

This reverts commit 9a28182b80c120a12e4e096b2676788b6e21d386.

Co-authored-by: Augustin <augustin@airbyte.io>

DoNotPanicUA committed:
Enable full SAT for the BigQuery source (#19720)

* enable full SAT for the BigQuery source

* use basic database instead of full due to SAT fails

* add full dataset sql

* minor changes

* specify catalogs (default works wrong)

* new line

* readme + format

Co-authored-by: Augustin <augustin@airbyte.io>

andriikorotkov committed:
added integration type field to the dectination_definition file and actor_definition table

arimbr committed:
Add notification and metrics docs to sidebar (#20210)

sophia-wiley committed:
typo fix (#20207)

erohmensing committed:
Add information about running docs live (#18914)

dizel852 committed:
🪟 🎨 New stream details panel (#19219)

* move all stream details table related components to separate folder;
create the first version of table powered by current implementation of react-table;

* init a copy of existing <Table /> component - <NextTable />

* fixed calc error

* first implementation of NextTable:
- use css modules instead of styled cmp
- extracted base table styles
- remove props used just for styling(customPadding, light, etc.)
- no typings

* remove StreamDetailsHeader

* updated StreamFieldsTable

* fix typings error

* update NextTable cells classnames

* remove last styled component from NextTable

* add base tests for NextTable component

* remove comments

* replace "any" type with "unknown" where possible

* add react-table v8 package

* update NextTable component

* update NextTable component

* extend react-table ColumnMeta interface

* update StreamFieldsTable component

* update NextTable tests

* update NextTable storybook

* fixed crashed StreamConnectionHeader during connection creation

* replace useConnectionEditService with useConnectionFormService hook

* remove getIcon wrap function and combine all parts in separate component

* update styles

* extract common logic to separate function

* fix performance

* fix not capitalized table header cell

* fix checkbox style issue

* make thead sticky by default

* fix not scrollable table

* fixed top bar overflow during scrolling table

* fix top bar overflow during scrolling table - second try

* fix hidden cursor and pk columns

* fix hidden cursor and pk columns

* fix PR comments: replace css values with variables

* remove unnecessary hook dependencies

* fix relative import component path

* add space around panel;
add 100px top gap height

* fix arrow color

* align table padding with toggle

* add ArrowRight icon from design
replace faArrowRight with ArrowRight

* move NextTable component to ui folder

* move StreamDetailsPanel component to separate folder

* move StreamPanelHeader component to separate folder

* remove obsolete DetailsPanel components

* move all DetailsPanel related components to separate folder

* fix when no namespace

* fixed details panel header sync mode label

* fix relative import path

* move react-table.d.ts to src/types

* remove fake destination data type column

* add FormattedMessage to ConnectorHeaderGroupIcon

* fix StreamPanelHeader style to match Figma

* change td color from grey-100 to grey-50
fix relative path

* temporary remove @tanstack/react-table package

* add @tanstack/react-table back to package.json

* fix cross icon color
fix items alignment in flex container

maxi297 committed:
[ISSUE #15628] apply lookback window on earliest datetime between sta… (#20156)

* [ISSUE #15628] apply lookback window on earliest datetime between start and cursor

* [ISSUE #15628] update release information and clean return statement

dizel852 committed:
🪟 🎨 Update toast design (#19980)

* feat: changed styling for toast

* fix: made changes requested

* fix: made changes requested

* update Toast design

* update Toast storybook

* replace "hasError" with actual type of Toast

* replace classes with mixins

* fix broken notifications

* change the animation to "ease-out"

* replace Notification type with interface

* make enum const

* use button dark theme

* fixed stretched icon

* move z-index value to css variables file

* fix toast bottom-margin and update animation

* add shadow mixin

* reduce spacing-page-bottom to 88px

* Revert "reduce spacing-page-bottom to 88px"

This reverts commit 6c72aced0130e19dbd4b36f82f519421cae371ef.

* extend base Button component: do not break the text

* align text to left
set baseline as flex-start

* set max width for notification container

* fix text center alignment

* fix action button margin

Co-authored-by: Harshith Mullapudi <harshithmullapudi@gmail.com>

mariamthiam committed:
:tada: Source Amplitude: enable data region selection (#19727)

* feat(Source Amplitude): enable data region selection

* fix(Source Amplitude): correct tests and url

* chore(Source Amplitude): use property for url_base

* fix(Source Amplitude): integration tests

* add backward to accpt test

* bump connector version and doc

* auto-bump connector version

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

hntan committed:
Source Gridly: add icon and make `grid_id` parameter required (#20048)

* Add source-gridly icon

* source-gridly mark grid_id as required field

* Update acceptance-test-config.yml

* Update acceptance-test-config.yml

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

benmoriceau committed:
Rm DbMigration Controller (#20213)

dizel852 committed:
add storybook for RadioButton component (#20228)

andriikorotkov committed:
fixed tests

andriikorotkov committed:
Merge remote-tracking branch 'origin/akorotkov/remove_factory' into akorotkov/remove_factory

# Conflicts:
#	airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java
#	airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/NormalizationRunnerFactory.java
#	airbyte-commons-worker/src/test/java/io/airbyte/workers/normalization/DefaultNormalizationRunnerTest.java

andriikorotkov committed:
fixed tests

andriikorotkov committed:
fixed tests
